src/main/java/webserver/domain/request/HttpRequest.java

```diff
@@ -0,0 +1,99 @@
+package webserver.domain.request;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import utils.IOUtils;
+import webserver.domain.Header;
+
+public class HttpRequest {
+    private static final String lineSeparator = System.getProperty("line.separator");
+    private static final Logger logger = LoggerFactory.getLogger(HttpRequest.class);
+
+    private final RequestLine requestLine;
+    private final Header header;
+    private final String body;
+
+    public HttpRequest(RequestLine requestLine, Header header, String body) {
+        this.requestLine = requestLine;
+        this.header = header;
+        this.body = body;
+    }
+
+    public HttpRequest(RequestLine requestLine, Header header) {
+        this.requestLine = requestLine;
+        this.header = header;
+        this.body = "";
+    }
+
+    public static HttpRequest of(BufferedReader br) throws IOException {
+        String line = br.readLine();
+        RequestLine requestLine = RequestLine.of(line);
+        logger.debug("Request Line : {}", line);
+
+        Map<String, String> headerFields = new HashMap<>();
+        while (!line.equals("")) {
+            line = br.readLine();
+            if (line == null) {
+                break;
+            }
+            String[] headerTokens = line.split(": ");
+            if (headerTokens.length >=2) {
+                headerFields.put(headerTokens[0], headerTokens[1]);
+            }
+            logger.debug("Header : {}", line);
+        }
+        Header header = new Header(headerFields);
+
+        String contentLength = headerFields.get("Content-Length");
+        if (Objects.isNull(contentLength)) {
+            return new HttpRequest(requestLine, header, "");
+        }
+        String body = IOUtils.readData(br, Integer.parseInt(headerFields.get("Content-Length")));
```

Comment:
```suggestion
        String body = IOUtils.readData(br, Integer.parseInt(contentLength));
```

사소하지만.. 해당 변수를 사용할 수 있을 것 같습니다 ㅎ_ㅎ